subsys/dfu/src/dfu_target_mcuboot.c

```diff
@@ -12,11 +12,46 @@
 
 LOG_MODULE_REGISTER(dfu_target_mcuboot, CONFIG_DFU_TARGET_LOG_LEVEL);
 
+#define MAX_FILE_SEARCH_LEN 500
 #define MCUBOOT_HEADER_MAGIC 0x96f3b83d
 
 static struct flash_img_context flash_img;
 static size_t offset;
 
+int dfu_ctx_mcuboot_set_b1_file(char *file, bool s0_active, char **update)
+{
+	if (file == NULL   ||  update == NULL) {
+		return -EINVAL;
+	}
+
+	/* Ensure that 'file' is null-terminated. */
+	if (strnlen(file, MAX_FILE_SEARCH_LEN) == MAX_FILE_SEARCH_LEN) {
+		return -EINVAL;
```

Comment:
IMO this should be a different error value from the NULL cases, since it is a valid failure case.

Reply:
Changed to "ENOTSUP"

Reply:
Please give a better alternative if this is not good enough :) I find it really hard to locate a good error code